File: app/src/main/java/me/zhanghai/android/files/ui/FixQueryChangeSearchView.java
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.ui;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class FixQueryChangeSearchView extends FixLayoutSearchView {

    private boolean mShouldIgnoreQueryChange;

    public FixQueryChangeSearchView(@NonNull Context context) {
        super(context);
    }

    public FixQueryChangeSearchView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public FixQueryChangeSearchView(@NonNull Context context, @Nullable AttributeSet attrs,
                                    int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public boolean shouldIgnoreQueryChange() {
        return mShouldIgnoreQueryChange;
    }

    @Override
    public void setIconified(boolean iconify) {
        mShouldIgnoreQueryChange = true;
        super.setIconified(iconify);
        mShouldIgnoreQueryChange = false;
    }

    @Override
    public void onActionViewCollapsed() {
        mShouldIgnoreQueryChange = true;
        super.onActionViewCollapsed();
        mShouldIgnoreQueryChange = false;
    }

    @Override
    public void onActionViewExpanded() {
        mShouldIgnoreQueryChange = true;
        super.onActionViewExpanded();
        mShouldIgnoreQueryChange = false;
    }
}

File: app/src/main/java/me/zhanghai/android/files/ui/FixQueryChangeSearchView.kt
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.ui

import android.content.Context
import android.util.AttributeSet
import androidx.annotation.AttrRes

class FixQueryChangeSearchView : FixLayoutSearchView {
    var shouldIgnoreQueryChange = false
        private set

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, @AttrRes defStyleAttr: Int) : super(
        context, attrs, defStyleAttr
    )

    override fun setIconified(iconify: Boolean) {
        shouldIgnoreQueryChange = true
        super.setIconified(iconify)
        shouldIgnoreQueryChange = false
    }

    override fun onActionViewCollapsed() {
        shouldIgnoreQueryChange = true
        super.onActionViewCollapsed()
        shouldIgnoreQueryChange = false
    }

    override fun onActionViewExpanded() {
        shouldIgnoreQueryChange = true
        super.onActionViewExpanded()
        shouldIgnoreQueryChange = false
    }
}
